### PR TITLE
Move more sprintf usages to snprintf

### PR DESCRIPTION
**Pull request by jkoritzinsky**
I missed a few the last time, but the runtime-sanitized leg caught them.

### CODE REVIEW

File: src/native/libs/System.Globalization.Native/pal_icushim.c [hunk 1]
```diff
@@ -417,7 +419,7 @@ static void InitializeUColClonePointers(char* symbolVersion)
     ucol_safeClone_ptr = (ucol_safeClone_func)GetProcAddress((HMODULE)libicui18n, symbolName);
 #else
     char symbolName[SYMBOL_NAME_SIZE];
-    sprintf(symbolName, "ucol_safeClone%s", symbolVersion);
+    snprintf(symbolName, SYMBOL_NAME_SIZE, "ucol_safeClone%s", symbolVersion);
```
Comment (tarekgh) [inline src/native/libs/System.Globalization.Native/pal_icushim.c:424]: >SYMBOL_NAME_SIZE [](http://example.com/codeflow?start=25&length=16)

I think this should be `MaxICUVersionStringLength + 1`

File: src/native/libs/System.Globalization.Native/pal_icushim.c [hunk 2]
```diff
@@ -446,7 +448,7 @@ static void InitializeVariableMaxAndTopPointers(char* symbolVersion)
     ucol_setVariableTop_ptr = (ucol_setVariableTop_func)GetProcAddress((HMODULE)libicui18n, symbolName);
 #else
     char symbolName[SYMBOL_NAME_SIZE];
-    sprintf(symbolName, "ucol_setVariableTop%s", symbolVersion);
+    snprintf(symbolName, SYMBOL_NAME_SIZE, "ucol_setVariableTop%s", symbolVersion);
```
Comment (tarekgh) [inline src/native/libs/System.Globalization.Native/pal_icushim.c:453]: >SYMBOL_NAME_SIZE [](http://example.com/codeflow?start=25&length=16)

this should be `MaxICUVersionStringWithSuffixLength + 1` too.

File: src/native/libs/System.Globalization.Native/pal_icushim.c [hunk 3]
```diff
@@ -538,7 +540,7 @@ void GlobalizationNative_InitICUFunctions(void* icuuc, void* icuin, const char*
 #if defined(TARGET_WINDOWS)
         sprintf_s(symbolSuffix, SYMBOL_CUSTOM_SUFFIX_SIZE, "_%s", suffix);
 #else
-        sprintf(symbolSuffix, "_%s", suffix);
+        snprintf(symbolSuffix, SYMBOL_CUSTOM_SUFFIX_SIZE, "_%s", suffix);
```
Comment (tarekgh) [inline src/native/libs/System.Globalization.Native/pal_icushim.c:545]: >SYMBOL_CUSTOM_SUFFIX_SIZE [](http://example.com/codeflow?start=31&length=25)

should this be `suffixAllowedSize`?

File: src/native/libs/System.Globalization.Native/pal_icushim.c (hunk 1 above)
Comment (jkoritzinsky) [reply src/native/libs/System.Globalization.Native/pal_icushim.c:424]: The array is defined above as `SYMBOL_NAME_SIZE` characters, so `SYMBOL_NAME_SIZE` is the right value here.

File: src/native/libs/System.Globalization.Native/pal_icushim.c (hunk 2 above)
Comment (jkoritzinsky) [reply src/native/libs/System.Globalization.Native/pal_icushim.c:453]: Same as the above case.

File: src/native/libs/System.Globalization.Native/pal_icushim.c (hunk 3 above)
Comment (jkoritzinsky) [reply src/native/libs/System.Globalization.Native/pal_icushim.c:545]: I think we should match the Windows case here as it's effectively the same bounds check.

File: src/native/libs/System.Globalization.Native/pal_icushim.c (hunk 1 above)
Comment (tarekgh) [reply src/native/libs/System.Globalization.Native/pal_icushim.c:424]: right. I overlooked that. please ignore.

File: src/native/libs/System.Globalization.Native/pal_icushim.c (hunk 3 above)
Comment (tarekgh) [reply src/native/libs/System.Globalization.Native/pal_icushim.c:545]: Fine with me if you want to keep it as Windows does it. `SYMBOL_CUSTOM_SUFFIX_SIZE` still safe.

**Review by tarekgh — APPROVED** [review]

**Comment by jkoritzinsky** [comment]
Failures are all known issues.

**Pull request merged**